test/api/v1/rpc_test.go

```diff
@@ -0,0 +1,176 @@
+package api_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/celestiaorg/talis/internal/db/models"
+	"github.com/celestiaorg/talis/pkg/api/v1/handlers"
+	"github.com/celestiaorg/talis/test"
+)
+
+var defaultProjectCreateParams = handlers.ProjectCreateParams{
+	Name:        "test-project",
+	Description: "A test project",
+	Config:      `{"resources": {"cpu": 2, "memory": "4GB"}, "settings": {"env": "test"}}`,
+}
+
+var defaultTaskGetParams = handlers.TaskGetParams{
+	ProjectName: "test-project",
+	TaskName:    "test-task",
+}
+
+func TestProjectRPCMethods(t *testing.T) {
+	suite := test.NewSuite(t)
+	defer suite.Cleanup()
+
+	// Create a project directly in the database
+	project := models.Project{
+		OwnerID:     models.AdminID,
+		Name:        defaultProjectCreateParams.Name,
+		Description: defaultProjectCreateParams.Description,
+		Config:      defaultProjectCreateParams.Config,
+	}
+	err := suite.ProjectRepo.Create(suite.Context(), &project)
+	require.NoError(t, err)
+
+	// Get project using RPC
+	getParams := handlers.ProjectGetParams{
+		Name: defaultProjectCreateParams.Name,
+	}
+	retrievedProject, err := suite.APIClient.GetProject(suite.Context(), getParams)
+	require.NoError(t, err)
+	// Don't check ID since it's auto-incremented by the DB
+	require.Equal(t, project.Name, retrievedProject.Name)
+	require.Equal(t, project.Description, retrievedProject.Description)
+	require.Equal(t, project.Config, retrievedProject.Config)
+
+	// List projects using RPC
+	listParams := handlers.ProjectListParams{
+		Page: 1,
+	}
+	projects, err := suite.APIClient.ListProjects(suite.Context(), listParams)
+	require.NoError(t, err)
+	require.NotEmpty(t, projects)
+	require.Equal(t, 1, len(projects))
+	// Don't check ID since it's auto-incremented by the DB
+	require.Equal(t, project.Name, projects[0].Name)
+
+	// Create another project
+	secondProject := models.Project{
+		OwnerID:     models.AdminID,
+		Name:        "second-project",
+		Description: "Another test project",
+		Config:      defaultProjectCreateParams.Config,
+	}
+	err = suite.ProjectRepo.Create(suite.Context(), &secondProject)
```

Comment:
Same, use the API client to create the project not the service directly.

Reply:
done